src/cr2header_int.cpp

```diff
@@ -17,7 +17,9 @@ namespace Exiv2 {
 
     bool Cr2Header::read(const byte* pData, uint32_t size)
     {
-        if (size < 16) return false;
```

Comment:
The `!pData` part is definitely correct. I am however not so sure about the later, as most other `XyzHeader` classes show a similar condition in their `read()` function. For me `size != 16` makes more sense, but maybe there is another reason?

Reply:
The specification is very clear about this, and the CR2 Header size must be 16:
http://lclevy.free.fr/cr2/#key_info

Furthermore, if you read the code of this function, you'll notice that it always tries to read 16 bytes (Except when if finds something wrong before, in that case it just return false).

Reply:
Probably in the other XyzHeader classes we have the same problem ... that's why is good to write tests for your code :yum:

Reply:
> The specification is very clear about this, and the CR2 Header size must be 16:
http://lclevy.free.fr/cr2/#key_info

Sure, I am quite certain that the other headers have similar specifications. My point is however, that the `Header::read` functions look like they should only fail if the specified size is less than the size of the header. For instance check out `src/crwedit.cpp:46`, there another header classes `read` function gets fed a `size` parameter that is the size of the whole input file (at least I think so after having a quick glance).

Your proposed change (which I like!) would unfortunately break that behavior.

Reply:
It's a valid point for the `CiffHeader` (The class being used in `src/crwedit.cpp:47`) but I checked the usages of `Cr2Header::read` and the only 2 usages are passing 16:

```
// cr2image.cpp:131
                if (0 == cr2Header.read(pData, 16)) {
// cr2image.cpp:244
        const int32_t len = 16;
        bool rc = header.read(buf, len);
```
Since it is a internal class we could even decide to change the signature of the function and remove the `uint32_t size` argument, since it is useless.

Reply:
We can certainly do that, but we have to leave the `length` parameter in the function signature as `cr2Header` inherits from `TiffHeaderBase` and `read` is a virtual function. If we remove the parameter it will no longer be callable via its base class.

Otherwise I fully agree with your proposition!